custom_components/waste_collection_schedule/waste_collection_schedule/source/banyule_vic_gov_au.py

```diff
@@ -0,0 +1,124 @@
+            geolocation_response = requests.get(
+                'https://www.banyule.vic.gov.au/api/v1/myarea/search',
+                params={
+                    'keywords': self._street_address,
+                    'maxresults': 1
+                }
+            )
+            geolocation_response.raise_for_status()
+
+            # Pull ID from results
+            geolocation_result = geolocation_response.json()
+            _LOGGER.debug(f"Search response: {geolocation_response!r}")
+
+            if 'success' in geolocation_result and not geolocation_result['success']:
+                raise SourceParseError('Unspecified server-side error when searching address')
+
+            if 'Items' not in geolocation_result or \
+                    geolocation_result['Items'] is None or \
+                    len(geolocation_result['Items']) < 1:
+                raise SourceParseError('Expected list of locations from address search, got empty or missing list')
+
+            geolocation_data = geolocation_result['Items'][0]
+
+            if 'Id' not in geolocation_data:
+                raise SourceParseError('Location in address search result but missing geolocation ID')
+
+            self._geolocation_id = geolocation_data['Id']
+            _LOGGER.info(f"Address {self._street_address} mapped to geolocation ID {self._geolocation_id}")
+
+        return self._geolocation_id
+
+    def fetch(self) -> typing.List[Collection]:
+        # Calendar lookup cares about a cookie, so a Session must be used
+        calendar_session = requests.Session()
+
+        calendar_request = calendar_session.get(
+            'https://www.banyule.vic.gov.au/Waste-environment/Waste-recycling/Bin-collection-services'
+        )
+        calendar_request.raise_for_status()
+
+        calendar_request = calendar_session.get(
+            'https://www.banyule.vic.gov.au/ocapi/Public/myarea/wasteservices',
+            params={
+                'geolocationid': self.geolocation_id,
+                'ocsvclang': 'en-AU'
+            }
+        )
+        calendar_request.raise_for_status()
+
+        calendar_result = calendar_request.json()
+        _LOGGER.debug(f"Calendar response: {calendar_result!r}")
+
+        if 'success' in calendar_result and not calendar_result['success']:
+            raise SourceParseError('Unspecified server-side error when getting calendar')
+
+        # Extract entries from bundled HTML
+        calendar_parser = BeautifulSoup(calendar_result['responseContent'], 'html.parser')
+
+        pickup_entries = []
+
+        for element in calendar_parser.find_all('article'):
+            _LOGGER.debug(f"Parsing collection: {element!r}")
+
+            waste_type = element.h3.string
+
+            # Extract and parse collection date
+            waste_date_str = element.find(class_='next-service').string.strip()
+            waste_date = datetime.strptime(waste_date_str.partition(' ')[2], '%d/%m/%Y')
```

Comment:
I think you'll want a tighter check here.

See #161 which is an open PR of mine to fix a bug that I started hitting with Stonnington's implementation this week.

Reply:
Yeah I saw your PR. For the short-term I don't think it's an issue since we don't have collections that haven't got a set date (at the moment) but it is a good sanity check regardless.

Reply:
That's how Stonnington worked until they broke it last week :wink: